src/java/org/apache/cassandra/journal/ActiveSegment.java

```diff
@@ -261,21 +264,34 @@ boolean shouldFlush()
      * TODO FIXME: calls from outside Flusher + callbacks
      * @return last synced offset
      */
-    synchronized int flush()
+    synchronized int flush(boolean fsync)
     {
         int allocatePosition = this.allocatePosition.get();
         if (lastFlushedOffset >= allocatePosition)
             return lastFlushedOffset;
 
         waitForModifications();
-        flushInternal();
+        if (fsync) flushInternal();
         lastFlushedOffset = allocatePosition;
+        lastFsyncOffsetUpdater.accumulateAndGet(this, allocatePosition, Math::max);
```

Comment:
is this correct?  if `fsync == false` then we didn't fsync... so why update the pointer?

Reply:
good catch, so much for late night updates (when will I learn)